src/los_client/cli.py

```diff
@@ -0,0 +1,158 @@
+import argparse
+import os
+import sys
+import json
+import subprocess
+import asyncio
+from dataclasses import dataclass
+from typing import Any
+
+@dataclass
+class CLIConfig:
+    json_path: str
+    solver_path: str = ""
+    output_path: str = ""
+    problem_path: str = ""
+
+    def load_config(self) -> None:
+
+        if os.path.exists(self.json_path):
+            try:
+                with open(self.json_path, "r") as config_file:
+                    config_json = json.load(config_file)
+                    self.solver_path = config_json.get("solver_path")
+                    self.output_path = config_json.get("output_path")
+                    self.problem_path = config_json.get("problem_path")
+            except (json.JSONDecodeError, IOError):
+                print("Warning: Failed to load configuration."
+                      " Creating new one.")
+                self.save_config()
+        else:
+            print("No existing configuration found. Creating new one.")
+            self.save_config()
+
+    def save_config(self) -> None:
+
+        config_json = {
+            "solver_path": self.solver_path,
+            "output_path": self.output_path,
+            "problem_path": self.problem_path,
+        }
+        try:
+            with open(self.json_path, "w") as config_file:
+                json.dump(config_json, config_file)
+        except IOError as e:
+            print(f"Error saving configuration: {e}", file=sys.stderr)
+
+@dataclass
+class SatCLI:
+    output_folder = os.path.join(os.path.dirname(__file__) , "output")
+    config : CLIConfig
+
+    def __init__(self) -> None:
+        self.config = CLIConfig(os.path.join(
+            self.output_folder, "config.json"))
+        self.config.load_config()
+
+    def configure(self, args:Any) -> None:
+
+        if args.solver:
+            if isinstance(args.solver, Dummy):
```

Comment:
What do you want to achieve with the Dummy thing? You can use hasattr to check if the argument was set if that is what you want.

Reply:
What I was aiming for is the user inputs -problem without an argument and it displays the current set path. If an argument is given then the path is overriden. I use the Dummy object to check which wether -problem or -solver or -output was used without an argument.

Reply:
Its better to use a subcommand for this, e.g., `los_client run` is for starting the client, while `los_client show` will show the current configuration. However, `los_client run` should not overwrite the stored configuration automatically. If you want to provide the ability to overwrite settings then we could have a command `los_client set`.